### DISCUSSION TITLE
How to format ISO 8601  Date for uploadDate - MediaObject C#

### DISCUSSION THREAD

**Discussion by nautilux2** [RehanSaeed/Schema.NET#121] [Q&A]
I don't know how to format datetime field to match the iso 8601 date in c#
I tried to use
`DateTime.Parse(info.PublishedDate.ToString("yyyy-MM-ddTHH:mm:ssZ"), null, System.Globalization.DateTimeStyles.RoundtripKind)` 
but when I @Html.Raw it procedure `"uploadDate":"2020-01-01"` why it must be `"uploadDate":"2020-01-01T01:01:01+07:00"`
I search many post but no luck
Thanks

**Comment by Turnerj** [comment]
The [`uploadDate`](https://schema.org/uploadDate) field on `MediaObject` is defined by schema.org as type [`Date`](https://schema.org/Date) (not [`DateTime`](https://schema.org/DateTime)). Schema.NET knows this distinction so when it serializes the date [it will serialize as `yyyy-MM-dd`](https://github.com/RehanSaeed/Schema.NET/blob/master/Source/Schema.NET/DateTimeToIso8601DateValuesJsonConverter.cs#L35). Essentially, this is a distinction by the schema.org that Schema.NET is following. This more limited format of `yyyy-MM-dd` is still a valid ISO 8601 date even without the timestamp.

Currently there isn't a way to skip this functionality though it could be a feature worth adding in the future. That said, other systems that interpret the field as the schema.org specification describes may not interpret the additional timestamp data you're wanting to provide.

**Comment by nautilux2** [comment]
Hi @Turnerj 
Thanks for your answer

I see in the metadata of MediaObject define the UploadDate
`public Values<int?, DateTime?> UploadDate { get; set; }`

Are there any difference between the ISO with the timestamp and without the timestamp when Google read structured data

**Comment by Turnerj** [comment]
I don't know exactly how Google will read it but if they follow the schema.org specifications for a `MediaObject`'s `uploadDate` field, they _should_ be expecting a date (`yyyy-MM-dd`). If it had the time too, it might include it or might just ignore the time component. I think it would be a pretty bad implementation on Google's side if they ignored the entire field because it was a date/time and not just a date.

It might be best to followup with Google specifically (though I don't know where you would go to do that) to see if they actually go beyond the specification and would use the time component if it was there.

If you are concerned that Google wouldn't read your field right, its probably best to leave it as the schema.org specification describes. I know it does suck a little especially if you've got a more specific date and you really want to display that.

**Comment by nautilux2** [comment]
I read the example of a recipe website in Google example and it provides date without timestamp
Anyway thanks for your answer one again and I will close this question
